v2/tools/generator/internal/config/object_model_configuration.go

```diff
@@ -0,0 +1,131 @@
+// ║ ObjectModelConfiguration ║───────│ GroupConfiguration │───────│ VersionConfiguration │───────│ TypeConfiguration │───────│ PropertyConfiguration │
+// ║                          ║1  1..n│                    │1  1..n│                      │1  1..n│                   │1  1..n│                       │
+// ╚══════════════════════════╝       └────────────────────┘       └──────────────────────┘       └───────────────────┘       └───────────────────────┘
+//
+type ObjectModelConfiguration struct {
+	groups map[string]*GroupConfiguration
+}
+
+// NewObjectModelConfiguration returns a new (empty) ObjectModelConfiguration
+func NewObjectModelConfiguration() *ObjectModelConfiguration {
+	return &ObjectModelConfiguration{
+		groups: make(map[string]*GroupConfiguration),
+	}
+}
+
+// TypeRename looks up a rename for the specified type, returning the new name and true if found, or empty string
+// and false if not.
+func (omc *ObjectModelConfiguration) TypeRename(name astmodel.TypeName) (string, bool) {
+	group, ok := omc.findGroup(name)
+	if !ok {
+		return "", false
+	}
+
+	return group.TypeRename(name)
+}
+
+// ARMReference looks up a property to determine whether it may be an ARM reference or not.
+func (omc *ObjectModelConfiguration) ARMReference(name astmodel.TypeName, property astmodel.PropertyName) (bool, bool) {
+	group, ok := omc.findGroup(name)
+	if !ok {
+		return false, false
+	}
+
+	return group.ARMReference(name, property)
+}
+
+// FindUnusedARMReferences returns a slice listing any unused ARMReference configuration
+func (omc *ObjectModelConfiguration) FindUnusedARMReferences() []string {
+	var result []string
+	for _, gc := range omc.groups {
+		result = append(result, gc.FindUnusedARMReferences()...)
+	}
+
+	sort.Strings(result)
+
+	return result
+}
+
+func (omc *ObjectModelConfiguration) Add(group *GroupConfiguration) *ObjectModelConfiguration {
+	if omc.groups == nil {
+		// Initialize the map just-in-time
+		omc.groups = make(map[string]*GroupConfiguration)
+	}
+	// store the group name using lowercase,
+	// so we can do case-insensitive lookups later
+	omc.groups[strings.ToLower(group.name)] = group
+	return omc
+}
+
+// findGroup uses the provided TypeName to work out which nested GroupConfiguration should be used
+func (omc *ObjectModelConfiguration) findGroup(name astmodel.TypeName) (*GroupConfiguration, bool) {
+	localRef, ok := name.PackageReference.AsLocalPackage()
+	if !ok {
+		return nil, false
+	}
+
+	group := strings.ToLower(localRef.Group())
+	g, ok := omc.groups[group]
+	if !ok {
+		return nil, false
+	}
+
+	return g, true
+}
+
+// UnmarshalYAML populates our instance from the YAML.
+// The slice node.Content contains pairs of nodes, first one for an ID, then one for the value.
+func (omc *ObjectModelConfiguration) UnmarshalYAML(value *yaml.Node) error {
+	if value.Kind != yaml.MappingNode {
+		return errors.New("expected mapping")
+	}
+
+	var lastId string
+	for i, c := range value.Content {
```

Comment:
minor: An alternative to this loop would've been:
```
for i :=0; i < len(value.Content); i +=2 {
    id := value.Content[i]
    // err handling here in case there is no i+1
    value := value.Content[i+1]
    // Handle stuff as you do below
}
```

I'm not really sure it's a clear win either way but figured I would mention it as the "multi-pass" loop that's building state with `%2` and `continue` is also a bit tricky to reason about and in some ways being able to pull those two pieces together into a single pass in the loop is slightly clearer I think (for me, YMMV).

Reply:
The advantages I see are primarily:

* `id` can be scoped inside the loop.
* fewer continues

This applies to all of the loops, but feel free to ignore it as it's not a huge deal.

Reply:
I'm not convinced the `+= 2` approach would be clearer.